pkg/kev/config/extensions_test.go

```diff
@@ -124,4 +171,43 @@ var _ = Describe("Extentions", func() {
 		})
 	})
 
+	Describe("Merge", func() {
+		It("merges target into base", func() {
+			k8sBase := config.DefaultK8SConfig()
+			var k8sTarget config.K8SConfiguration
+			k8sTarget.Workload.Replicas = 10
+			k8sTarget.Workload.LivenessProbe.Type = config.ProbeTypeNone.String()
+
+			expected := k8sBase
+			expected.Workload.Replicas = 10
+			expected.Workload.LivenessProbe.Type = config.ProbeTypeNone.String()
+
+			result, err := k8sBase.Merge(k8sTarget)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(BeEquivalentTo(expected))
+		})
+
+		When("configs are empty/invalid", func() {
+			It("returns default when map is empty", func() {
+				extensions := make(map[string]interface{})
+				parsedConf, err := config.K8SCfgFromMap(extensions)
+				Expect(err).NotTo(HaveOccurred())
+
+				result, err := config.DefaultK8SConfig().Merge(parsedConf)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(result).To(BeEquivalentTo(config.DefaultK8SConfig()))
+			})
+
+			It("returns default when map is nil", func() {
+				var extensions map[string]interface{}
+				parsedConf, err := config.K8SCfgFromMap(extensions)
```

Comment:
it's hard to see from this spec as to what value is meant to be nil? Does the `config.K8SCfgFromMap` ever returns nil? Having a separate context with the stage prep in `BeforeEach` where input nil value is defined would be great for legibility.